### PR TITLE
117 fix export with ifs

### LINKED ISSUE #117: Fix export with IFS
## Example

```bash
minishell$ export A="a b c"
minishell$ echo $A
a
```

```bash
minishell$ export A="a a" # OK
minishell$ export A="a 1" # NG
minishell: export: `1': not a valid identifier
minishell$ export A="1 2 3 4" #NG
minishell: export: `2': not a valid identifier
minishell: export: `3': not a valid identifier
minishell: export: `4': not a valid identifier
```

### PR DESCRIPTION
**Pull request by alex216**
fix #117 

## Summary by CodeRabbit

- **New Features**
	- Introduced new functionalities for expanding variables specifically for heredocs.
	- Added capabilities for normalizing lists and splitting strings by specific delimiters.

- **Bug Fixes**
	- Updated variable expansion logic to enhance performance and accuracy, particularly in heredoc scenarios.

- **Documentation**
	- Updated relevant documentation to reflect changes in function names and signatures.

- **Tests**
	- Enhanced test coverage for new features and updated existing tests to align with function renaming.

### CODE REVIEW

File: include/expansion.h [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 t_list	*expand(t_list *cmd_args, t_ctx *ctx);
 
-char	*expand_variable(char *str, t_ctx *ctx, bool is_smart_expand);
+char	*expand_variable_for_heredoc(char *str, t_ctx *ctx);
```
Comment (rask24) [inline include/expansion.h:22]: Is `expand_variable_heredoc` a more appropriate naming?

**Review by rask24 — COMMENTED** [review]
@alex216 Please check the comments!
I've confirmed the program works properly 👍🏼 

Overall, I've noticed variables with names ending in '_'.
It seems this is used to avoid name collisions.
But I think it may be better practice to consider alternative naming for local variables that doesn't conflict with function arguments, rather than using this convention(ending variables with _).
This approach could lead to more contextually appropriate and clearer variable names.

File: include/expansion.h (hunk 1 above)
Comment (alex216) [reply include/expansion.h:22]: True.

Then, is `exec/wait_children.c` more appropriate than`exec/wait_for_children.c` too?

Comment (rask24) [reply include/expansion.h:22]: I think so 👍🏼

Comment (rask24) [reply include/expansion.h:22]: I'll handle renaming `wait_for_children` in #137

**Comment by alex216** [comment]
- [ ] variable inside double quote should not recognize IFS
```bash
minishell$ export A="a b"
minishell$ cd "$A"
minishell: cd: too many arguments
minishell$ bash
bash-5.2$
bash-5.2$ cd "$A"
bash: cd: a b: No such file or directory
bash-5.2$
```

**Review by rask24 — APPROVED** [review]
@alex216 LGTM 🇺🇬

**Pull request merged**